### ISSUE TITLE
Setting "startval" to null breaks display_required_only

### ISSUE THREAD

**Issue by mortenscheel** [json-editor/json-editor#1632]
### General information

- json-editor version: @latest

### Expected behavior
With `{display_required_only: false}` the form should also show optional fields.

### Actual behavior
If the options also include `startval: null` optional fields are hidden.
I suspect that other options are affected too, but I haven't confirmed it yet.

### Steps to reproduce the behavior
In the linked example, update the form data with either `null` or an empty form:
![image](https://github.com/user-attachments/assets/50446184-c262-45fb-9615-c7498a059a1a)

Direct link to example: https://json-editor.github.io/json-editor/?data=N4Ig9gDgLglmB2BnEAuUBXRBTA+vAhgLa75RQBOMARulFsihelgDQgTliHQ5VYBmYcrgAmWADZY6qJqxCZRA/OnFQcAN3zjmDWW0L4AHjjHQAFqgAMbGmQQ5EUUrkJgxqAIxsAxvmwOsJBhYdVwOSCxyKABPAPxybwtGcmY2RDMwAHccSM5yBhAYeDpyfG9YBBA2cO41CD86HANjSXgAcygzHGEysywRVH4tbDZhV1CcIfF/cIhI2HpB4blAwQTcbwRHJem5SDUinEyYETapVBBErG8AayowQyrCzfhxagvBYvxM+i4sAFYnmAqAAra5qcT4aJgWgXeBCAziJ6IK4GVCgAAkKL6aJQIDMZAgiBQAHoSSDEAgALTYrAGAB0QjaJJEpX4UBJACZLNyqR5OSTaWi2DE5hdgWDyk9ZvMYIs0CACMR0SBRVgLo5KO0nrAoJILgA5Ijq/RFAAygQ6SQ8AF82HT8DAkQq1RqKEU2jrgvq8QBRKkGJ0gG12kDCACO6BgwgGKAA2orjSAALqhzp09V4+5gKCa/AQAAswZtQA=

```javascript
{
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "title": "Name",
      "minLength": 1
    },
    "email": {
      "type": "string",
      "title": "E-mail"
    }
  },
  "required": [
    "name"
  ]
}
```

**Comment by schmunk42** [comment]
TL;DR that's what `show_opt_in` is for.

In your case with a `string` (Email) illustrates the problem, if you show an empty input, do you want an empty string or nothing as key/value?

**Comment by mortenscheel** [comment]
@schmunk42 no that's not the issue. I'll do my best to clarify, but please ask if there's something that isn't clear.

Without an explicit `startval`. This works as expected. The email field is displayed even though it's optional.
```js
new JSONEditor(element, {
    // schema: The example schema,
    display_required_only: false
});
```
With `startval` explicitly set to `null` (which is the default value according to the readme), the email field is hidden.
```js
new JSONEditor(element, {
    // schema: The example schema,
    display_required_only: false,
    startval: null
});
```
Since the default value of `startval` is `null`, I find it surprising that including `startval: null` has any effect _at all_.
Now I'm aware of it this gotcha, so I can just omit startval when appropriate, but I assumed that it's a bug, which is why I posted this issue.

Explicitly setting `startval` to `null` in the options has other surprising effects too. E.g. if the schema is an array of objects without a minItems requirement, the editor will usually just show a plus button initially. But with `startval: null`, it starts with a sub-item editor open.

**Comment by schmunk42** [comment]
The README might be misleading here @germanbisurgi Should we remove the default value for `startval` or rather use `undefined`?

When a `startval` is used, `setValue()` is called ad the editor behaves differently.
When there's no `startval` the editor shows all fields initially, unless `display_required_only: true`.
When `startval` or `setValue()` is used `display_required_only` has no effect.

@germanbisurgi This is something for the docs.

**Comment by silvester-pari** [comment]
We stumbled upon this as well, we tried to debug why some form fields were missing, and ultimately found out that using `startval` or `setValue()` does some unexpected things. Even doing a `setValue(null)` or `setValue(undefined)` (see minimal reproduction [here](https://pmk65.github.io/jedemov2/dist/demo.html?schema=EQbwOgdgBFbALgTwA4FM4C5bAPYCMArVAY3jgBpIY54BLeAG3WCzgGEBDAJwquy9QBHAK60BAE0xQA2n2rAAZjhxw5AXUrRsyLjjRc6qAM5TwW%2BUpUsoZmHexI0UuEfhdaEAOarzUAL6a9nB43KZydjQozKzAru5ePvZ%2BkHzJEH7AQA%3D&value=ETI%3D&code=EQehAIBUAsFNwGYHsA2KkHcCWA7A5uCrrAM7gCGATvALbk4Am5ALkpQJ4WPjXkNI4U7AHTgAmkgCu4AMb0KDBrMklWNWUgbxyAIyQA3bdx2x0GYQB0cWBOAAUAKxKwGWVpXC5V9GbCS3sRkxhACkAZQB5ADkAUVd3AEpwJxc3NmEtVUokdjsEgG4rFPi2cABecBxYDHBA%2FnNw6Li0yjt%2BGUkaWBxmYQBHSVgOMNNYGXc7C2AAYicBAFpU93nkShophIAaZJcWcgKrQ5xiluEBSeBeBnYp7bzygD5wAG8rcHBnZkgsLqlmO3uZSerxw73eJ3cwk%2BADVyChBnZJIxYAhiAwEm9wABfbYAJgADISMTgsQlgEA%3D%3D&style=ETI%3D&theme=bootstrap3&iconlib=bootstrap3&object_layout=normal&template=default&show_errors=interaction&required_by_default=0&no_additional_properties=0&display_required_only=0&remove_empty_properties=0&keep_oneof_values=0&ajax=0&ajaxCredentials=0&show_opt_in=0&disable_edit_json=0&disable_collapse=0&disable_properties=0&disable_array_add=0&disable_array_reorder=0&disable_array_delete=0&enable_array_copy=0&array_controls_top=0&disable_array_delete_all_rows=0&disable_array_delete_last_row=0&prompt_before_delete=0&lib_aceeditor=0&lib_autocomplete=0&lib_sceditor=0&lib_simplemde=0&lib_select2=0&lib_selectize=0&lib_choices=0&lib_flatpickr=0&lib_signaturepad=0&lib_mathjs=0&lib_cleavejs=0&lib_jodit=0&lib_jquery=0&lib_dompurify=1)) strangely hides optional fields, even though they are rendered initially if no value is given. I would give a +1 to the impression that this is rather a bug than unclear documentation - I would assume if the initial value is undefined, it should behave the same as if the value is **set** to undefined.

Happy to know more about what the reasoning behind this is, maybe we missed something!
